appenders/bash/logger.sh

```diff
@@ -1,52 +1,59 @@
 #!/bin/bash
 # Dependencies: curl
 # Usage:
+# ```
+# source logger.sh
+# pulse_debug "debug message"
+# pulse_info "info message"
+# pulse_warn "warn message"
+# pulse_trace "trace message"
+# ```
 # Export the variables: PULSE_COLLECTOR_HOST and PULSE_COLLECTOR_PORT for the host and port caontaing the Pulse
 # Log Collector
 # ```
-# export PULSE_COLLECTOR_HOST=host1.com
-# export PULSE_COLLECTOR_PORT=9004
+# export PULSE_LOG_COLLECTOR_HOST=valhalla.phdata.io
+# export PULSE_LOG_COLLECTOR_PORT=9004
+# export APPLICATION_NAME=pulse_test_default
 # ```
-# Then source this scrip in your own bash scrip
-# ``` 
-# source logger.sh
-# logger --category=category --level=level --message=message --threadname=threadname --application=application-name
-# ```
-set -euo pipefail
+
+declare ERROR_MESSAGE="Error! Message info is only possible for its argument."
 
 logger(){
-  if [ "$#" -lt 1 ]
+  # $1 = Level Info
+  # $2 = the number of funtion parameters
+  # $3 = Message to be logged
+  # $4 = Pulse Logging text
+  if [ -z "$APPLICATION_NAME" ] || [ -z "$PULSE_LOG_COLLECTOR_HOST" ] || [ -z "$PULSE_LOG_COLLECTOR_PORT" ]
   then
-  echo "Usage Info: logger --category=category --level=level --message=message --threadname=threadname --application=application-name"
-  exit 1
+    echo "Error! Env variable should have PULSE_LOG_COLLECTOR_HOST, PULSE_LOG_COLLETOR_PORT, and APPLICATION_NAME"
+  else
+    if [ "$2" -ne 1 ]
+    then
+      echo $ERROR_MESSAGE
+    else
+      timestamp=$( date -u +"%Y-%m-%dT%H:%M:%SZ")
+      if [ $PULSE_DEBUG ]
+      then
+        echo $timestamp
+        echo "Pulse is "$4" \"$3\" message"
+      fi
+      curl -s -X POST -H 'Content-Type: application/json' -d '{"category": "","timestamp": "'$timestamp'", "level": "'$1'", "message": "'$3'", "threadName": ""}' http://$PULSE_LOG_COLLECTOR_HOST:$PULSE_LOG_COLLECTOR_PORT/v2/event/$APPLICATION_NAME
+    fi
   fi
+}
 
-  while [ $# -gt 0 ];do
-  case "$1" in
-    --category=*)
-        category="${1#*=}"
-    ;;
-    --level=*)
-        level="${1#*=}"
-    ;;
-    --message=*)
-        message="${1#*=}"
-    ;;
-    --threadName=*)
-        threadName="${1#*=}"
-    ;;
-    --application=*)
-        application="${1#*=}"
-        ;;
-    *)
-        echo "Usage Info: logger --category=category --level=level --message=message --threadname=threadname"
-        exit 1
-  esac
-    shift
-  done
-
-  timestamp=$( date -u +"%Y-%m-%dT%H:%M:%SZ")
+pulse_debug(){
+  logger "debug" "$#" "$1" "debugging"
+}
 
-  curl -s -X POST -H 'Content-Type: application/json' -d '{"category": "'$category'","timestamp": "'$timestamp'", "level": "'$level'", "message": "'$message'", "threadName": "'$threadName'"}' http://$PULSE_COLLECTOR_HOST:$PULSE_COLLECTOR_PORT/v2/event/$application
+pulse_warn(){
+  logger "warn" "$#" "$1" "warning"
```

Comment:
why is `$#` needed?

Reply:
@afoerster , `$#` is the number of function arguments, and `pulse_***` functions should have only one argument, the message to be logged, as I explained above.